app/admin/flags.rb

```diff
@@ -87,9 +87,14 @@ def dismiss
     end
 
     def remove_project_submission
-      resource.project_submission.destroy
+      result = Admin::Flags::RemoveSubmission.call(current_user, flag: resource)
 
-      redirect_to admin_flags_path, notice: 'Success: Submission has been removed.'
+      if result.success?
+        resource.project_submission.discard
```

Comment:
Could we move this into the `RemoveSubmission` service please?

Reply:
Yep! That makes more sense there.